table/block_based/partitioned_filter_block.cc

```diff
@@ -304,6 +305,7 @@ Status PartitionedFilterBlockReader::GetFilterPartitionBlock(
   }
 
   ReadOptions read_options;
+  read_options.rate_limiter_priority = rate_limiter_priority;
```

Comment:
I feel like the root cause of this issue is that `GetFilterPartitionBlock()` was pretending to have a proper `ReadOptions`. I think it would be better if we could plumb the operation's (e.g., `Get()`'s) `ReadOptions` down into `GetFilterPartitionBlock()` and use it here. Is it possible?

Reply:
In https://github.com/facebook/rocksdb/pull/10251 , we have passed the rate_limiter_priority from BlockBasedTable to *MayMatch* functions. I missed passing rate_limiter_priority into GetFilterPartitionBlock. 

For ReadOptions param:
1. I roughly remember that we just wanted to pass a light-weight rate_limiter_priority rather than ReadOptions into the filter functions.
2. The filter functions may not need all the options from ReadOptions of the caller (e.g. Get()).

cc @pdillinger

Reply:
I agree it makes sense to refactor at some point to pass the operation's ```ReadOptions``` down to all the filter functions. But not all the options may be appropriate, so we eventually have to filter out the unnecessary ones before calling ```RetrieveBlock```.